### ISSUE TITLE
Attempting to get agent key right after registering it will occasionally fail

### ISSUE THREAD

**Issue by rossengeorgiev** [wazuh/wazuh-api#39]
Hi,
I've been encountering a race condition between registering the agent and immediately trying to get the key. I assume this happens since `authd` will take a bit of time to write the file, but `wazuh-api` just read `client.keys` to get the key. That leads the the following situation:

```bash
$ curl -u foo:bar -k -X POST -d 'name=NewAgent&ip=10.0.0.8' https://API_IP:55000/agents
{"error":0,"data":"001"}
$ curl -u foo:bar -k -X GET https://API_IP:55000/agents/001/key
{"error":1701,"message":"Agent does not exist: 001"}
```

I could wait between the two calls but that is not a good solution. Alternative, perhaps the API can return the key in the response when adding an agent and make the whole process a single API call.

Not using `authd` is not an option due to #26

**Comment by jesuslinares** [comment]
It is added to the roadmap.

Thank you so much.

**Comment by mgmacias95** [comment]
Hello @rossengeorgiev,

We've just improved our API calls to add agents to return both ID and key (see https://github.com/wazuh/wazuh/pull/303) . This new functionality is awaiting to be merged and will be available in Wazuh 3.1.

Best regards,
Marta
